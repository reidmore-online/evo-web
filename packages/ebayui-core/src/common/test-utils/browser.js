import { fireEvent, createEvent } from "@marko/testing-library";
/**
 * Simulates a touch based scroll event over 4 animation frames.
 *
 * @param {HTMLElement} el The element to scroll.
 * @param {number} to The new scrollLeft for the element.
 * @param {function} cb A callback to call after the scroll.
 */
function simulateScroll(el, to, cb) {
    fireEvent.scroll(el);
    el.scrollLeft = to;
    setTimeout(cb, 600);
}

function waitFrames(count, cb) {
    if (count) {
        return requestAnimationFrame(() => {
            waitFrames(count - 1, cb);
        });
    }

    cb();
}

async function pressKey(el, info) {
    for (const event of [
        createEvent.keyDown(el, info),
        createEvent.keyUp(el, info),
    ]) {
        // we assign properties to them for older browsers (chrome 49)
        Object.keys(info).forEach((key) => {
            if (event[key] !== info[key]) {
                Object.defineProperty(event, key, { value: info[key] });
            }
        });

        await fireEvent(el, event);
    }
}

const fastAnimations = {
    // Adds an style to the document which forces all transitions to run more quickly for the tests.
    start(disable) {
        if (this.fastAnimationStyle) {
            return;
        }
        const time = disable ? 0 : 1;

        this.fastAnimationStyle = document.createElement("style");
        this.fastAnimationStyle.innerHTML = `*,details::details-content,*::before,*::after {
                animation-duration: ${time}ms !important;
                transition-duration: ${time}ms !important;
                transition-delay: ${time}ms !important;
                scroll-behavior: auto !important;
            }`;
        document.head.appendChild(this.fastAnimationStyle);
    },
    stop() {
        if (!this.fastAnimationStyle) {
            return;
        }

        document.head.removeChild(this.fastAnimationStyle);
        this.fastAnimationStyle = undefined;
    },
};

export { fastAnimations, waitFrames, pressKey, simulateScroll };
