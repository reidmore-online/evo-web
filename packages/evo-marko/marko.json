{
  "taglib-id": "evo-marko",
  "tags-dir": "./src/tags",
  "taglib-imports": [
    "./src/tags/evo-icon/marko.json",
    "./src/tags/evo-skeleton/marko.json"
  ]
}
