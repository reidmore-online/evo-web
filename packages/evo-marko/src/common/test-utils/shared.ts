export function getNItems(n: number, getAttrs: (i: number) => any[]) {
  return Array.from({ length: n }).map((_, i) => getAttrs(i));
}

// Helper method to render pagination items
export function getPaginationItems(
  n: number,
  href: boolean = false,
  selected: number = -1,
  navDisabled: boolean = false,
  skipPrevNext: boolean = false,
) {
  const items = [];
  if (!skipPrevNext) {
    items.push({
      type: "previous",
      href: href ? "#" : null,
      disabled: navDisabled,
    });
  }
  for (let i = 0; i < n; i++) {
    items.push({
      href: href ? "#" : null,
      current: i === selected,
    });
  }
  if (!skipPrevNext) {
    items.push({
      type: "next",
      href: href ? "#" : null,
      disabled: navDisabled,
    });
  }
  return items;
}
