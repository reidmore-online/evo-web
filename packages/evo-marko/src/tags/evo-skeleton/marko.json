{
  "taglib-id": "evo-marko-skeleton",
  "tags-dir": "./tags"
}
