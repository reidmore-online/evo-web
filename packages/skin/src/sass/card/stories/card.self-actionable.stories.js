export default { title: "Skin/Card/Self Actionable" };

export const verticalButton = () => `
<button class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            Professional-grade rookie cards featuring future Hall of Famers in their awkward rookie photo phase.
        </div>
    </div>
</button>
`;

export const verticalLink = () => `
<a class="card" href="https://ebay.com">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            A collection proving that somewhere between a curveball, a three-pointer, and a slap shot lies the secret to happiness.
        </div>
    </div>
</a>
`;

export const horizontalButton = () => `
<button class="card card--horizontal">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            Professional-grade rookie cards featuring future Hall of Famers in their awkward rookie photo phase.
        </div>
    </div>
</button>
`;

export const horizontalLink = () => `
<a class="card card--horizontal" href="https://ebay.com">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            A collection proving that somewhere between a curveball, a three-pointer, and a slap shot lies the secret to happiness.
        </div>
    </div>
</a>
`;
