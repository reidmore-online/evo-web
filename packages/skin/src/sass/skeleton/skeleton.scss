@use "../variables/variables";
@use "../mixins/private/button-mixins";
@use "../mixins/private/token-mixins";

@keyframes skeleton-fade-out {
    from {
        opacity: 1;
    }
    to {
        opacity: 0;
    }
}
@keyframes skeleton-fade-in {
    from {
        opacity: 0;
    }
    to {
        opacity: 1;
    }
}

::view-transition-new(skeleton-transition-*) {
    animation: skeleton-fade-in var(--motion-duration-medium-2)
        var(--motion-easing-continuous);
}
::view-transition-old(skeleton-transition-*) {
    animation: skeleton-fade-out var(--motion-duration-short-3)
        var(--motion-easing-continuous);
}

.skeleton {
    container-name: skeleton-container;
    container-type: inline-size;
    cursor: progress;
    width: 100%;
}

.skeleton__avatar,
.skeleton__button,
.skeleton__text,
.skeleton__image,
.skeleton__textbox {
    @include token-mixins.background-token(
        skeleton-background,
        color-loading-fill
    );

    width: 100%;
}

/* TODO: Need to swap these with component css variables and expose it instead */
.skeleton__avatar {
    border-radius: var(--avatar-border-radius, 50%);
    height: 48px;
    width: 48px;
}

.skeleton__button {
    border-radius: var(
        --btn-border-radius,
        calc(button-mixins.$button-height-regular / 2)
    );
    height: button-mixins.$button-height-regular;
}

.skeleton__button--small {
    border-radius: var(
        --btn-border-radius,
        calc(button-mixins.$button-height-small / 2)
    );
    height: button-mixins.$button-height-small;
}

.skeleton__button--large {
    border-radius: var(
        --btn-border-radius,
        calc(button-mixins.$button-height-large / 2)
    );
    height: button-mixins.$button-height-large;
}

.skeleton__text {
    border-radius: var(--text-border-radius, 3px);
    height: 16px;
    width: calc(100% - var(--spacing-300));
}

.skeleton__text::after {
    height: 16px;
    margin-top: calc(16px + var(--spacing-100));
}

.skeleton__text--large {
    height: 24px;
}

.skeleton__text--large::after {
    height: 24px;
    margin-top: calc(24px + var(--spacing-100));
}

.skeleton__text--multiline {
    margin-bottom: var(--spacing-300);
    position: relative;
    width: calc(100% - var(--spacing-300));
}

.skeleton__text--multiline::after {
    background: inherit;
    content: "";
    position: absolute;
    width: calc(100% - var(--spacing-700));
}

.skeleton__text--large.skeleton__text--multiline {
    margin-bottom: var(--spacing-400);
}

.skeleton__textbox {
    @include token-mixins.border-radius-token(
        textbox-border-radius,
        border-radius-50
    );

    height: 48px;
}

.skeleton__image {
    @include token-mixins.border-radius-token(
        image-border-radius,
        border-radius-50
    );

    height: 100%;
    max-width: initial;
}

/* TODO: Use aspect-ratio : 1 / 1 to set as square when we have browser support, till then */
/* using pseudo element to set as square */
.skeleton__image::after {
    bottom: 0;
    height: 100%;
    left: 0;
    position: absolute;
    right: 0;
    top: 0;
    width: 100%;
}

span.skeleton__avatar,
span.skeleton__button,
span.skeleton__text,
span.skeleton__image,
span.skeleton__textbox {
    display: inline-block;
}

span.skeleton__avatar:not(:last-child),
span.skeleton__button:not(:last-child),
span.skeleton__text:not(:last-child),
span.skeleton__image:not(:last-child),
span.skeleton__textbox:not(:last-child) {
    margin-inline-end: var(--spacing-100);
}

div.skeleton__avatar:not(:last-child),
div.skeleton__button:not(:last-child),
div.skeleton__text:not(:last-child),
div.skeleton__image:not(:last-child),
div.skeleton__textbox:not(:last-child) {
    margin-block-end: var(--spacing-150);
}

.skeleton--on-secondary {
    --skeleton-background: var(--color-loading-fill-on-secondary);
}

.skeleton--purple {
    --skeleton-background: var(--color-gradient-ai-purple-subtle);
}

.skeleton--green {
    --skeleton-background: var(--color-gradient-ai-green-subtle);
}

.skeleton--blue {
    --skeleton-background: var(--color-gradient-ai-blue-subtle);
}

@media (prefers-reduced-motion: no-preference) {
    @keyframes on-primary {
        0% {
            background-color: var(--color-loading-first);
        }
        100% {
            background-color: var(--color-loading-second);
        }
    }

    @keyframes on-secondary {
        0% {
            background-color: var(--color-loading-on-secondary-first);
        }
        100% {
            background-color: var(--color-loading-on-secondary-second);
        }
    }

    .skeleton__avatar,
    .skeleton__button,
    .skeleton__text,
    .skeleton__image,
    .skeleton__textbox {
        animation: var(--motion-duration-long-3) var(--motion-easing-continuous)
            infinite alternate on-primary;
    }

    .skeleton--on-secondary .skeleton__avatar,
    .skeleton--on-secondary .skeleton__button,
    .skeleton--on-secondary .skeleton__text,
    .skeleton--on-secondary .skeleton__image,
    .skeleton--on-secondary .skeleton__textbox {
        animation: var(--motion-duration-long-3) var(--motion-easing-continuous)
            infinite alternate on-secondary;
    }
}

/* A progressive enhancement to make the skeleton image rounded */
@container skeleton-container (width > 79px) {
    .skeleton__image {
        @include token-mixins.border-radius-token(
            image-border-radius,
            border-radius-100
        );
    }
}

[dir="rtl"] {
    .skeleton__text--multiline::after {
        left: var(--spacing-700);
    }
}
