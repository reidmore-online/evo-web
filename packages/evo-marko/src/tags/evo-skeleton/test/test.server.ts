import { describe, it } from "vitest";

import { composeStories } from "@storybook/marko";
import { snapshotHTML } from "../../../common/test-utils/snapshots";
import * as stories from "../skeleton.stories";
const { Default, Avatar, Tile, Text, TextMultiLine } = composeStories(stories);

describe("skeleton", () => {
    it("renders default skeleton", async () => {
        await snapshotHTML(Default);
    });

    it("renders avatar", async () => {
        await snapshotHTML(Avatar);
    });

    it("renders tile", async () => {
        await snapshotHTML(Tile);
    });

    it("renders text", async () => {
        await snapshotHTML(Text);
    });

    it("renders text multiLine", async () => {
        await snapshotHTML(TextMultiLine);
    });
});
