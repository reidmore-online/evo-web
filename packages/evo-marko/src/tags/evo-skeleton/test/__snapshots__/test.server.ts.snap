// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`skeleton > renders avatar 1`] = `
<DocumentFragment>
  <div
    class="skeleton"
    role="img"
  >
    <div
      class="skeleton__avatar"
    />
  </div>
</DocumentFragment>
`;

exports[`skeleton > renders default skeleton 1`] = `
<DocumentFragment>
  <div
    aria-label="Loading"
    class="skeleton"
    role="img"
    style="width: 220px"
  />
</DocumentFragment>
`;

exports[`skeleton > renders text 1`] = `
<DocumentFragment>
  <div
    class="demo-text-container"
  >
    <div
      class="skeleton"
      role="img"
    >
      <div
        class="skeleton__text"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`skeleton > renders text multiLine 1`] = `
<DocumentFragment>
  <div
    class="demo-text-container"
  >
    <div
      class="skeleton"
      role="img"
      style="width:220px"
    >
      <div
        class="skeleton__text skeleton__text--multiline skeleton__text--large"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`skeleton > renders tile 1`] = `
<DocumentFragment>
  <div
    class="demo-tile-container"
  >
    <div
      class="skeleton"
      role="img"
    >
      <div
        class="skeleton__image custom-image"
      />
      <div
        class="skeleton__text skeleton__text--multiline skeleton__text--large"
      />
    </div>
  </div>
</DocumentFragment>
`;
